api/src/services/import-services/import-critters-service.ts

```diff
@@ -0,0 +1,411 @@
+    const taxonomy = await this.platformService.getTaxonomyByTsns(critterTsns);
+
+    return taxonomy.map((taxon) => taxon.tsn);
+  }
+
+  /**
+   * Get a mapping of collection units for a list of tsns.
+   * Used in the zod validation.
+   *
+   * @example new Map([['Population Unit', new Set(['Atlin', 'Unit B'])]]);
+   *
+   * @async
+   * @param {WorkSheet} worksheet - Xlsx Worksheet
+   * @param {string[]} tsns - List of unique and valid TSNS
+   * @returns {Promise<Map<string, ICollectionUnitWithCategory[]>} Collection unit mapping
+   */
+  async _getCollectionUnitMap(worksheet: WorkSheet, tsns: string[]) {
+    const collectionUnitMap = new Map<string, { collectionUnits: ICollectionUnitWithCategory[]; tsn: number }>();
+
+    const collectionUnitColumns = this._getNonStandardColumns(worksheet);
+
+    // If no collection unit columns return empty Map
+    if (!collectionUnitColumns.length) {
+      return collectionUnitMap;
+    }
+
+    // Get the collection units for all the tsns in the worksheet
+    const tsnCollectionUnits = await Promise.all(
+      tsns.map((tsn) => this.critterbaseService.findTaxonCollectionUnits(tsn))
+    );
+
+    tsnCollectionUnits.forEach((collectionUnits, index) => {
+      if (collectionUnits.length) {
+        collectionUnitMap.set(toUpper(collectionUnits[0].category_name), { collectionUnits, tsn: Number(tsns[index]) });
+      }
+    });
+
+    return collectionUnitMap;
+  }
+
+  /**
+   * Validate CSV worksheet rows against reference data.
+   *
+   * @async
+   * @param {number} surveyId - Survey identifier
+   * @param {WorkSheet} worksheet - Xlsx worksheet
+   * @returns {Promise<Validation>} Conditional validation object
+   */
+  async _validateRows(surveyId: number, worksheet: WorkSheet): Promise<Validation> {
+    const rows = this._getRows(worksheet);
+    const nonStandardColumns = this._getNonStandardColumns(worksheet);
+
+    // Retrieve the dynamic validation config
+    const [validRowTsns, surveyCritterAliases] = await Promise.all([
+      this._getValidTsns(worksheet),
+      this.surveyCritterService.getUniqueSurveyCritterAliases(surveyId)
+    ]);
+    const collectionUnitMap = await this._getCollectionUnitMap(worksheet, validRowTsns);
+
+    // Parse reference data for validation
+    const tsnSet = new Set(validRowTsns.map((tsn) => Number(tsn)));
+    const csvCritterAliases = rows.map((row) => row.animal_id);
+
+    // Track the row validation errors
+    const errors: ValidationError[] = [];
+
+    const csvCritters = rows.map((row, index) => {
+      /**
+       * --------------------------------------------------------------------
+       *                      STANDARD ROW VALIDATION
+       * --------------------------------------------------------------------
+       */
+
+      // SEX is a required property and must be a correct value
+      const invalidSex = !row.sex || !CSV_CRITTER_SEX_OPTIONS.includes(toUpper(row.sex));
```

Comment:
This does a non-case-sensitive validation check in SIMS, but CritterBase's zod validation fails:

```
 {                                                                                                              
   received: 'male',                                                                                            
   code: 'invalid_enum_value',                                                                                  
   options: [Array],                                                                                            
   path: [Array],                                                                                               
   message: "Invalid enum value. Expected 'Male' | 'Female' | 'Unknown' | 'Hermaphroditic', received 'male'"    
 }                                                                                                              
```

Reply:
Also, does the sex options enum in SIMS need to include `'Hermaphroditic'`?

Reply:
Nice catch. Fixing